### DISCUSSION TITLE
How to set `swap`'s state?

### DISCUSSION THREAD

**Discussion by duskmoon314** [saadeghi/daisyui#1054] [Q&A] [answered]
I'm trying to use `swap` as a theme toggle in my website's header. But I can't figure out how to set its state according to `theme` in the `localStorage`. Thanks for any help.

---

I'm using Astro. Here is part of my current code:

```Astro
---
...
const theme = localStorage.getItem("theme")
---

...
<label class="btn btn-ghost btn-circle swap swap-rotate">
  <input
    type="checkbox"
    checked={theme === "night"}
    data-toggle-theme="night,winter"
  />
  <Icon class="swap-on w-6" name="icon-park-outline:moon" />
  <Icon class="swap-off w-6" name="icon-park-outline:sun-one" />
</label>
...
```

I have tried the snippet in #808. It seems not working either:

```Astro
<label class="btn btn-ghost btn-circle swap swap-rotate">
  <input type="checkbox" checked={theme === "night"} />
  <i
    class="w-6 h-6 iconify swap-on"
    data-icon="tabler:sun"
    data-set-theme="winter"
    >light
  </i>
  <i
    class="w-6 h-6 iconify swap-off"
    data-icon="tabler:moon"
    data-set-theme="night"
    >dark
  </i>
</label>
```

I just check the `checked` property of the checkbox and it seems nothing happens when it is clicked.

**Comment by saadeghi** [comment, marked as answer]
Are you hydrating your component? because the variable that reads the local storage must run on browser (not at build time) and it looks like your using an Astro component (or a page itself) which [can not be hydrated](https://docs.astro.build/en/core-concepts/framework-components/#can-i-hydrate-astro-components).  
Use hydratable UI component (React, Preact, Svelte, Vue, SolidJS, AlpineJS, Lit.) and hydrate it using `client:load` so the code runs on browser.

**Reply by duskmoon314** [reply]
Reasonable, I will give it a try. Thanks!

**Reply by duskmoon314** [reply]
I wrap this in a svelte component. It works like a charm. Thanks!
